### PR TITLE
Add config.map entry for Spack build

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: config.map [hunk 1]
```diff
@@ -1,4 +1,5 @@
 SCRAM_ARCH=el8_amd64_gcc11;PKGTOOLS_TAG=V00-34-XX;CMSDIST_TAG=IB/CMSSW_13_0_X/master;RELEASE_BRANCH=master;RELEASE_QUEUE=CMSSW_13_0_X;DOCKER_IMG=cmssw/el8;BUILD_PATCH_RELEASE=1;ADDITIONAL_TESTS=HLT,static-checks,fwlite,igprof-mp,igprof-pp,iwyu,lizard,flawfinder,check-headers,cfipython,invalid-includes,profiling,crab;PR_TESTS=1;PROD_ARCH=1;PRS_TEST_CLANG=1;REQUIRED_TEST=true;
+SCRAM_ARCH=el8_amd64_gcc11;PKGTOOLS_TAG=V00-34-XX;CMSDIST_TAG=IB/CMSSW_13_0_X/master;RELEASE_BRANCH=master;RELEASE_QUEUE=CMSSW_13_0_SPACK_X;DOCKER_IMG=cmssw/el8;BUILD_HOUR=23;BUILD_DAY=2;SPACK=1
```
Comment (smuzaffar) [inline config.map:2]: missing `;` at the end. You should also add `DISABLED=1;` otherwise normal build process will try to build a SPACK based release

Comment (iarspider) [reply config.map]: @smuzaffar `;` added, thanks. I have added 
https://github.com/cms-sw/cms-bot/pull/1880/files#diff-5953d6405892c14ec84e02b260d33a48dd172343f0cdf2af0d0fa71144aea950L55-R55 to stop regular scripts from picking up spack release.

File: config.map [hunk 2]
```diff
@@ -1,4 +1,5 @@
 SCRAM_ARCH=el8_amd64_gcc11;PKGTOOLS_TAG=V00-34-XX;CMSDIST_TAG=IB/CMSSW_13_0_X/master;RELEASE_BRANCH=master;RELEASE_QUEUE=CMSSW_13_0_X;DOCKER_IMG=cmssw/el8;BUILD_PATCH_RELEASE=1;ADDITIONAL_TESTS=HLT,static-checks,fwlite,igprof-mp,igprof-pp,iwyu,lizard,flawfinder,check-headers,cfipython,invalid-includes,profiling,crab;PR_TESTS=1;PROD_ARCH=1;PRS_TEST_CLANG=1;REQUIRED_TEST=true;
+SCRAM_ARCH=el8_amd64_gcc11;PKGTOOLS_TAG=V00-34-XX;CMSDIST_TAG=IB/CMSSW_13_0_X/master;RELEASE_BRANCH=master;RELEASE_QUEUE=CMSSW_13_0_SPACK_X;DOCKER_IMG=cmssw/el8;BUILD_HOUR=23;BUILD_DAY=2;SPACK=1;
```
Comment (smuzaffar) [inline config.map:2]: @iarspider , instead of adding a new line just add `SPACK=1;` in the existing line above

File: build-cmssw-ib-with-patch [hunk 3]
```diff
@@ -52,7 +52,7 @@ esac
 CMSDIST_REPO=cms-sw
 PKGTOOLS_REPO=cms-sw
 unset BUILD_PATCH_RELEASE
-eval `cat $CMS_BOT_DIR/config.map | grep -v 'DISABLED=' | grep "SCRAM_ARCH=$ARCHITECTURE;" | grep "RELEASE_QUEUE=$RELEASE_QUEUE;"`
+eval `cat $CMS_BOT_DIR/config.map | grep -v 'DISABLED=' | grep -v 'SPACK=' | grep "SCRAM_ARCH=$ARCHITECTURE;" | grep "RELEASE_QUEUE=$RELEASE_QUEUE;"`
```
Comment (smuzaffar) [inline build-cmssw-ib-with-patch:55]: this should be removed once you add `SPACK=1;` in the existing entry

File: config.map (hunk 2 above)
Comment (iarspider) [reply config.map:2]: Ah, that's a much better approach!

**Pull request merged**